microservices/services/query-metric/service/src/main/java/datawave/microservice/querymetric/handler/QueryMetricCombiner.java

```diff
@@ -204,6 +205,14 @@ public T combineMetrics(T updatedQueryMetric, T cachedQueryMetric, QueryMetricTy
             if (combinedMetric.getPredictions() == null && updatedQueryMetric.getPredictions() != null) {
                 combinedMetric.setPredictions(updatedQueryMetric.getPredictions());
             }
+            // Check to see if we should be adding or setting the subPlans to combineMetric
```

Comment:
It might help to use comments fully describe the if else logic here.

Reply:
Improved the combining logic in the latest commit and included comments.